transcom/mymove
# MAIN B-20452 - Office User - Edit, Submit, Approve, Delete Boat Shipment

## [B-20452](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20452)

NOTE: This should get merged in after this Main PR: https://github.com/transcom/mymove/pull/13618

INT PR: 
https://github.com/transcom/mymove/pull/13666
https://github.com/transcom/mymove/pull/13695

REBASE INT PR: https://github.com/transcom/mymove/pull/13816

## Summary

This is to make sure the edit, delete, submit, and approve (as SC/TOO) functionality for boat shipments works and also added fixes for issues found during unit testing.

 - Fixed the issue where the edit page for boat shipment was not displaying.
 - Added testharness for boat haul-away shipment needing SC and needing TOO approval.
 - Added playwright tests for edit, delete, submit boat shipment as SC.
 - Added playwright tests for edit, delete, approve boat shipment as TOO.
 - Fixed console warnings for props required.
 - Refactored code for checking shipment incomplete.
 - Fixed issue where you can still continue and sign move as a customer when boat or mobile home shipment is incomplete.

### How to test

1. Use `BoatHaulAwayMoveNeedsSC` testharness to create a boat shipment move.
3. As a SC -> find the move you have created -> Click `Edit shipment` and edit values and Save -> Make sure the changed values saved.
4. Click `Submit move details` in Move details tab -> After confirming submit, make sure you see `Move submitted` message.
5. As a TOO -> find the move -> Click the checkbox for boat shipment in Move Details tab and click `Approve selected` and then `Approve and send` to approve the shipment.
6. Verify it was approved in Move details tab.

Checking incomplete issue:
1. As a customer -> create a move and select boat for shipment and only fill out the 1st page of the form and click continue to go to 2nd page.
2. From the 2nd page of the form, click `Back` button 2 times to go back to the main move page.
3. Check that the boat shipment shows incomplete and `Review and submit` button is disabled.

## Screenshots
![20452-1](https://github.com/user-attachments/assets/8d423c63-a212-4172-8396-a2514d2df4f1)
![20452-2](https://github.com/user-attachments/assets/49e7fa8b-7f65-4425-8e26-edcc6acb692a)
![20452-3](https://github.com/user-attachments/assets/a7041c2f-31b4-4987-9ac8-1ccdad395438)
![20452-4](https://github.com/user-attachments/assets/d134aef4-3cc5-49ff-9c27-025ab19d9d6e)
